File: schedule/src/main/java/com/metting/scheduling/schedule/models/Agendamento.java
package com.metting.scheduling.schedule.models;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.metting.scheduling.schedule.enums.SituacaoAgendamento;
import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.Date;
import java.util.List;

@Getter
@Setter
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "agendamento")
public class Agendamento {

    @Id
    @Column(name = "agendamentoid")
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seq_agendamento")
    @SequenceGenerator(name = "seq_agendamento", sequenceName = "seq_agendamento", allocationSize = 1)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "usuarioid", referencedColumnName = "usuarioid",  insertable = false, updatable = false)
    private Usuario usuario;

    @NotNull(message = "É necessário informar o usuário que realizou o agendamneto.")
    @Column(name = "usuarioid")
    private Long usuarioId;

    @Column(name = "titulo")
    @NotNull(message = "Campo título obrigatório")
    private String titulo;

    @Column(name = "observacao")
    private String observacao;

    @Column(name = "datahora")
    @Temporal(TemporalType.TIMESTAMP)
    @JsonFormat
            (shape = JsonFormat.Shape.STRING, pattern = "dd-MM-yyyy HH:mm:ss")
    @NotNull(message = "Data e horário do agendamento é obrigatório.")
    private Date dataHora;

    @Column(name = "datahorainicio")
    @Temporal(TemporalType.TIMESTAMP)
    @JsonFormat
            (shape = JsonFormat.Shape.STRING, pattern = "dd-MM-yyyy HH:mm:ss")
    private Date dataHoraInicio;

    @Column(name = "datahorafim")
    @Temporal(TemporalType.TIMESTAMP)
    @JsonFormat
            (shape = JsonFormat.Shape.STRING, pattern = "dd-MM-yyyy HH:mm:ss")
    private Date dataHoraFim;

    @Column(name = "realizado")
    private Boolean realizado;

    @Column(name = "situacao")
    @Enumerated(EnumType.STRING)
    private SituacaoAgendamento situacao;

    @Column(name = "nota")
    private Integer nota;

    @Column(name = "feedback")
    private String feedback;

    @OneToMany(mappedBy = "agendamento", cascade = CascadeType.ALL)
    @JsonManagedReference(value = "agendamento_reference")
    private List<AgendamentoParticipante> participantes;
}

File: schedule_app/lib/models/agendamento.dart
import 'package:enum_to_string/enum_to_string.dart';
import 'package:schedule_app/enumators/agemdanto-status.dart';
import 'package:schedule_app/models/participante.dart';

class Agendamento {
  int id;
  int usuarioId;
  String titulo;
  String observacao;
  dynamic dataHora;
  dynamic dataHoraInicio;
  dynamic dataHoraFim;
  bool realizado;
  AgendamentoStatus situacao;
  List<Participante> participantes;
  int nota;
  String feedback;

  Agendamento();

  Agendamento.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    usuarioId = json['usuarioId'];
    titulo = json['titulo'];
    observacao = json['observacao'];
    dataHora = json['dataHora'];
    dataHoraInicio = json['dataHoraInicio'];
    dataHoraFim = json['dataHoraFim'];
    realizado = json['realizado'];
    situacao = EnumToString.fromString(AgendamentoStatus.values, json['situacao']);
    participantes = List();
    json['participantes'].forEach((e) {
      participantes.add(Participante.fromJson(e['participante']));
    });
    nota = json['nota'];
    ;
    feedback = json['feedback'];
  }

  Map<String, dynamic> toJson() {
    Map<String, dynamic> data = Map();
    data['id'] = id;
    data['usuarioId'] = usuarioId;
    data['titulo'] = titulo;
    data['observacao'] = observacao;
    data['dataHora'] = dataHora;
    data['dataHoraInicio'] = dataHoraInicio;
    data['dataHoraFim'] = dataHoraFim;
    data['relizado'] = realizado;
    if (situacao != null) {
      data['situacao'] = situacao.toString();
    }
    data['participantes'] = [];

    participantes.forEach((element) {
      data['participantes'].add(AgedamentoParticipante(participante: element).toJson());
    });

    data['nota'] = nota;
    data['feedback'] = feedback;
    return data;
  }
}

class AgedamentoParticipante {
  Null id;
  Null agendamento;
  Participante participante;

  AgedamentoParticipante({this.id, this.agendamento, this.participante});

  AgedamentoParticipante.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    agendamento = json['agendamento'];
    participante = json['participante'] != null ? new Participante.fromJson(json['participante']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['agendamento'] = this.agendamento;
    if (this.participante != null) {
      data['participante'] = this.participante.toJson();
    }
    return data;
  }
}
